backend/src/apiserver/storage/pipeline_store.go

```diff
@@ -425,3 +447,350 @@ func (s *PipelineStore) UpdatePipelineAndVersionsStatus(id string, status model.
 func NewPipelineStore(db *DB, time util.TimeInterface, uuid util.UUIDGeneratorInterface) *PipelineStore {
 	return &PipelineStore{db: db, time: time, uuid: uuid}
 }
+
+func (s *PipelineStore) CreatePipelineVersion(v *model.PipelineVersion) (*model.PipelineVersion, error) {
+	newPipelineVersion := *v
+	newPipelineVersion.CreatedAtInSec = s.time.Now().Unix()
+	id, err := s.uuid.NewRandom()
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create a pipeline version id.")
+	}
+	newPipelineVersion.UUID = id.String()
+
+	// Prepare queries of inserting new version and updating default version.
+	versionSql, versionArgs, versionErr := sq.
+		Insert("pipeline_versions").
+		SetMap(
+			sq.Eq{
+				"UUID":           newPipelineVersion.UUID,
+				"CreatedAtInSec": newPipelineVersion.CreatedAtInSec,
+				"Name":           newPipelineVersion.Name,
+				"Parameters":     newPipelineVersion.Parameters,
+				"PipelineId":     newPipelineVersion.PipelineId,
+				"Status":         string(newPipelineVersion.Status),
+				"CodeSourceUrl":  ""}).
```

Comment:
why empty string?

Reply:
... Thanks for the catch! Forgot about this place after the initial quick draft. Added unit test including this field.